doc/api/vm.md

```diff
@@ -459,47 +457,56 @@ const bar = new vm.SourceTextModule(`
 //
 // "Link" the imported dependencies of this Module to it.
 //
-// The provided linking callback (the "linker") accepts two arguments: the
-// parent module (`bar` in this case) and the string that is the specifier of
-// the imported module. The callback is expected to return a Module that
-// corresponds to the provided specifier, with certain requirements documented
-// in `module.link()`.
-//
-// If linking has not started for the returned Module, the same linker
-// callback will be called on the returned Module.
+// Obtain the requested dependencies of a SourceTextModule by
+// `sourceTextModule.moduleRequests` and resolve them.
 //
 // Even top-level Modules without dependencies must be explicitly linked. The
-// callback provided would never be called, however.
-//
-// The link() method returns a Promise that will be resolved when all the
-// Promises returned by the linker resolve.
+// array passed to `sourceTextModule.linkRequestedModules(modules)` can be
+// empty, however.
```

Comment:
Shall we not just automatically do this at this point? This could pave the way for even allowing it to be optional even for modules with dependencies under certain linking mode configurations where you could opt-in to the the host fallback on the context say.